<?xml version="1.0"?>
<launch>

  <!-- Define the control and sim arguments -->
  <arg name="controller_type"  default= "Lyap" />         <!-- Lyap, PD, PI, PID fuzzyPD-->
  <arg name="path_type"        default= "lemniscate" />  <!-- ellipse, lemniscate, line, sine, rectangle , spiral-->
  <arg name="guardar_datos"    default= "True" />       <!-- /parametro para guardar o no datos en un txt -->
  
  <param name="r"   type = "double"    value="0.0485"/>   <!-- wheel radius -->
  <param name="lx"   type = "double"    value="0.0975"/>  <!-- distance in cm along the x axis from center between the two wheels to one of the wheels -->
  <param name="ly"   type = "double"    value="0.103"/>   <!-- distance in cm along the y axis from center between the two wheels to one of the wheels -->
  
  <param name="path_type"     type="str" value="$(arg path_type)" />
  <param name="guardar_datos" type="bool" value="$(arg guardar_datos)" />
  
  
  <!-- Load trajectory parameters -->
  <rosparam file="$(find jetauto_trajectory_control)/config/trajectory_params.yaml" command="load" />
  <!-- Load controller parameters -->
  <rosparam file="$(find jetauto_trajectory_control)/config/controller_params.yaml" command="load" />
  <!-- Load FIS -->
  <param name="path_json" type="string" value="$(find jetauto_trajectory_control)/config/Mandamijson.json" />
  



  <!-- Launch case -->
   <group if="$(eval controller_type == 'fuzzyPD')">
   <node name="controller_cinem_fuzzyPD" pkg="jetauto_trajectory_control" type="fuzzyPD.py" output="screen" />
  </group>


  <group if="$(eval controller_type == 'PD')">
   <node name="controller_cinem_PD" pkg="jetauto_trajectory_control" type="controller_cinem_PD.py" output="screen" />
  </group>

  <group if="$(eval controller_type == 'PI')">
    <node name="controller_cinem_PI" pkg="jetauto_trajectory_control" type="controller_cinem_PI.py" output="screen" />
  </group>

  <group if="$(eval controller_type == 'Lyap')">
    <node name="controller_cinem_Lyap" pkg="jetauto_trajectory_control" type="controller_cinem_Lyap.py" output="screen" />
  </group>

  <group if="$(eval controller_type == 'PID')">
    <node name="controller_pid" pkg="jetauto_trajectory_control" type="controller_cinem_PID.py" output="screen" />
  </group>

  <node name="Pose_Estimation" pkg="jetauto_pose_estimation" type="jetauto_pose_estimation.py" output="screen"/>
  <include file="$(find jetauto_driver_speed)/launch/jetauto_driver_speed.launch"/>
  
  <include file="$(find jetauto_descriptionV2)/launch/display.launch"/> 

</launch>

